File: frontend/src/lib/schemas/dataset.ts
import { z } from "zod";

export const DatasetResponseSchema = z.object({
  id: z.string().uuid(),
  project_id: z.string().uuid(),
  filename: z.string(),
  schema_version: z.string(),
  row_count: z.number(),
  status: z.enum(["uploaded", "validated", "failed"]),
  created_at: z.string(),
});

export type DatasetResponse = z.infer<typeof DatasetResponseSchema>;

File: backend/app/schemas/dataset.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DatasetResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    filename: str
    schema_version: str
    row_count: int
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class SourceExampleResponse(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    project_id: uuid.UUID
    external_id: str | None
    source_hash: str
    payload: dict
    created_at: datetime

    model_config = {"from_attributes": True}
